File: backend/app/db/models/project.py
"""Project model."""
from datetime import datetime
from sqlalchemy import Column, String, Text, DateTime, JSON
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func, text

from app.db.database import Base


class Project(Base):
    """Project model for organizing work."""
    
    __tablename__ = "projects"
    
    id = Column(UUID(as_uuid=True), primary_key=True, server_default=text("gen_random_uuid()"))
    name = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    settings = Column(JSON, nullable=True)  # RAG config: chunk_size, top_k, etc.
    created_at = Column(DateTime, server_default=func.now(), nullable=False)
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    
    # Relationships
    documents = relationship("Document", back_populates="project", cascade="all, delete-orphan")
    tasks = relationship("Task", back_populates="project", cascade="all, delete-orphan")
    chat_messages = relationship("ChatMessage", back_populates="project", cascade="all, delete-orphan")
    slack_integration = relationship("SlackIntegration", back_populates="project", uselist=False, cascade="all, delete-orphan")

File: frontend/src/types/project.ts
export interface Project {
  id: string;
  name: string;
  description: string | null;
  settings: Record<string, any> | null;
  created_at: string;
  updated_at: string | null;
}

export interface ProjectSettings {
  chunk_size?: number;
  chunk_overlap?: number;
  top_k?: number;
  similarity_threshold?: number;
}

export interface ProjectCreate {
  name: string;
  description?: string;
  settings?: ProjectSettings;
}

export interface ProjectUpdate {
  name?: string;
  description?: string;
  settings?: ProjectSettings;
}

export interface ProjectListResponse {
  projects: Project[];
  total: number;
}
